plumbing/bootstrap.py

```diff
@@ -74,88 +67,79 @@ def clean_error_message(message):
     return message
 
 
-def create_datapackage(description_file):
+def create_datapackage(description_file, source_folder):
     """Convert description.source.yaml into datapackage.json."""
 
-    datapackage_file = description_file.replace('yaml', 'json')
+    datapackage_file = os.path.join(source_folder, DATAPACKAGE_FILE)
 
     with open(description_file) as stream:
         try:
             package = yaml.load(stream.read())
             package['name'] = slugify(package['title'], separator='_')
             package['extraction_processors'] = select_processors(package)
-        except ParserError as error:
+        except (ParserError, ScannerError) as error:
             message = clean_error_message(str(error))
             package = {'yaml_error': message}
             logging.debug('Parser error in %s: %s', description_file, message)
 
     with open(datapackage_file, 'w+') as stream:
         stream.write(json.dumps(package, indent=4))
 
-    logging.debug('Created %s', datapackage_file)
+    pipeline_id = os.path.split(source_folder)[-1]
+    logging.info('%s: created datapackage', pipeline_id)
 
     return package
 
 
-def bootstrap_pipeline(source_folder):
+def bootstrap_pipeline(package, source_folder):
     """Save the pipeline file inside the source folder."""
 
     _, source_name = os.path.split(source_folder)
-    package = load_source_datapackage(source_folder)
     pipeline = {source_name: DEFAULT_PIPELINE}
 
     if 'yaml_error' not in package:
```

Comment:
@cyberbikepunk as per my previous review, I still do not understand what looks to me like a strange swallowing and silencing of an exception, and saving it as data, to use again only to silently pass on it.